### ISSUE TITLE
RandomForestClassifier only works with int

### ISSUE THREAD

**Issue by skjerns** [emlearn/emlearn#9]
Thank's to #8 I can finally compile the code :)

Now I see that the exporting seems to work only with features trained on `int`, is that correct?

Is there any way to easily adapt this to work natively with `float/double`? 

```Python
import numpy as np
import sklearn_porter
import matplotlib.pyplot as plt
import emlearn
from sklearn.ensemble import RandomForestClassifier


train_x = np.random.rand(1000, 8) # input is float64
train_y = np.random.randint(0, 4, 1000)
clf = RandomForestClassifier(n_estimators=51)
clf.fit(train_x, train_y)
    
cmodel = emlearn.convert(clf)  
cmodel.predict(train_x) # predicts the same number all over again as train_x is float
```

edit:
 I've already found a hack by rescaling `float [0->1]` to `int [0->2147483647]`:
in `trees.py` I change line 27 to
```Python
            n = [ feature, int(th*2147483647), left, right ]
```

now predict gives the same result
```Python
pred1 = cmodel.predict(train_x*2147483647) # predicts the correct number
pred2 = clf.predict(train_x)
np.allclose(pred1, pred2)
```

However this seems kind of hacky and a native support might be better (also in case of floats outside the range `0->1`)

**Comment by jonnor** [comment]
Yeah float should be the primary data format, with uint32/16/8 as the secondary.

I expect to get around to that next week

**Comment by skjerns** [comment]
great! if I can be of any help let me know.

**Comment by jonnor** [comment]
@skjerns can you test if this works for you now in master/0.9.0?

**Comment by skjerns** [comment]
Thanks very much :)

It seems to be majorly working!

However, I seem to have the same problem as in https://github.com/nok/sklearn-porter/issues/52 .
(I assume it's the same issue as I have the exact same label predictions wrong as when using sklearn-porter. Using `m2cgen` it works fine.)

I think there is some inconsistency when calculating the final probability with a tree that is not fully grown (`max_depth!=None`). The simple calculation of winning classes per tree works in most cases, but the actual computations depends on the depth of the trees and therefore deviates in a few border cases, where the actual probabilities per tree need to be calculated and averaged.
